tests/unit/tabular/README.md

```diff
@@ -0,0 +1,146 @@
+
+    Returns:
+        dict: A tabular representation of the resource.
+    
+    Pseudo Code:
+        1. get the resourceType from resource, lookup config, raise exception if not found
+        2. get the editable columns from config
+        3. flatten the resource, remove keys not in editable columns
+        4. return the flattened resource
+    """
+    pass    
+```
+
+
+#### from_tabular
+
+This naive pseudo-Python code defines a function `from_table` that takes a tabular object as input and merges it with the original resource.
+
+see https://pypi.org/project/deepmerge/
+
+```python
+def from_tabular(tabular: dict, config: Config, project_id: str, resource: dict) -> dict:
+    """Convert a tabular representation of a resource to a resource. (mock for testing)
+
+    Args:
+        tabular (dict): A tabular (edited) representation of a resource.
+        config (Config): A configuration object.
+        project_id (str): The project ID.
+        resource (immutable dict): The original resource.
+
+    Returns:
+        dict: A merged resource
+
+    Pseudo Code:
+        1. get the resourceType from resource, lookup config, raise exception if not found
+        2. get the editable columns from config, raise exception if tabular contains extraneous keys
+        3. unflatten the resource, 
+        4. return merger of the original resource and the unflattened tabular
+        
+    """
+    pass
+```
+
+### validation
+
+The validation module contains functions that validate a resource against a schema.
+
+```python
+def validate(resource_dict: dict) -> ParseResult:
+    """Validate a resource. (mock for testing)
+
+    Args:
+        resource_dict (dict): A resource in JSON format.
+
+    Returns:
+        ParseResult: The validated resource.
+        
+    Pseudo Code:
+        1. raise exception if resource_dict does not contain a resourceType
+        2. validate the resource against the schema for the resourceType
+        3. return a parse result that contains the validated resource and any validation exception 
+        
+    """
+    pass
+
+```
+
+
+### workflow
+
+The tabular methods above need to be incorporated into a command line workflow.
+The documentation for these commands should be maintained in the [user facing documentation here](https://github.com/ACED-IDP/aced-idp.github.io/blob/0b9b33dc0542795e756219ef8305dd2e33bcc173/docs/workflows/metadata.md#L20-L19) 
+
+
+Pseudo Code:
+    1. Download [all|subset]existing metadata from the portal e.g. `gen3_util meta cp --profile=aced --project_id=aced-test <PATH-TO-FHIR>`
+    2. Convert the FHIR data to tabular form. e.g. `gen3_util meta to_tabular <PATH-TO-FHIR> <PATH-TO-TABULAR>`
+    3. Edit the tabular data
+    4. Convert the tabular data to FHIR and validate. e.g. `gen3_util meta from_tabular <PATH-TO-TABULAR> <PATH-TO-FHIR>`
+    5. Upload the metadata to the portal e.g. `gen3_util meta publish --profile=aced --project_id=aced-test <PATH-TO-FHIR>` 
```

Comment:
Minor formatting change, looks like Github doesn't render numbered lists with indents. Using the updated markdown below sets each numbered item on it's own line.

## Current List

![Screenshot 2023-12-20 at 12 04 17 PM](https://github.com/ACED-IDP/gen3_util/assets/15309567/30d444d9-95ac-4c4a-99f7-0236459a175b)

## Updated Markdown

```md
Pseudo Code:
    1. Download [all|subset]existing metadata from the portal e.g. `gen3_util meta cp --profile=aced --project_id=aced-test <PATH-TO-FHIR>`
    2. Convert the FHIR data to tabular form. e.g. `gen3_util meta to_tabular <PATH-TO-FHIR> <PATH-TO-TABULAR>`
    3. Edit the tabular data
    4. Convert the tabular data to FHIR and validate. e.g. `gen3_util meta from_tabular <PATH-TO-TABULAR> <PATH-TO-FHIR>`
    5. Upload the metadata to the portal e.g. `gen3_util meta publish --profile=aced --project_id=aced-test <PATH-TO-FHIR>` 
```

## Updated List

![Screenshot 2023-12-20 at 12 05 38 PM](https://github.com/ACED-IDP/gen3_util/assets/15309567/b7842e62-8498-4091-b2e7-2a0d83e544de)

Reply:
done.